src/ir_interpreter/interpret_ir.ml

```diff
@@ -498,16 +498,14 @@ and interpret_exp_mut env exp (k : V.value V.cont) =
     interpret_exp env' exp1 k
   | TryE (exp1, cases, Some (id, ty)) ->
     assert env.flavor.has_await;
-    let exp2 = Construct.(varE (var id ty)) in
+    let exp2 = Construct.(varE (var id ty) -*- unitE ()) in
```

Comment:
@crusso suggested to not construct syntax on the fly, but interpret the call. Not sure if that is simpler, yet.